src/main/java/com/internet/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.internet.shop.service.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ShoppingCartService;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        List<Product> products = shoppingCart.getProducts();
+        products.add(product);
+        shoppingCart.setProducts(products);
+        return shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        List<Product> products = shoppingCart.getProducts();
+        boolean remove = products.remove(product);
+        shoppingCart.setProducts(products);
+        shoppingCartDao.update(shoppingCart);
```

Comment:
Optionally: If you don't mind stream API usage, we could simplify this code a lot.
But it's up to you.

Reply:
we can remove the product from the list by removeIf, but this will change only the local list. So we will need to create and update the list in the shopping cart and DB anyway. So it will not make it easier